src/main/scala/fly/play/jiraExceptionProcessor/Jira.scala

```diff
@@ -1,124 +1,147 @@
-              throw new Exception("Problem retrieving the 'Hash' custom field ('%s'): %s" format (status, response.body))
-          }
+  private lazy val hashCustomField =
+    // request a list of all fields
+    request("field").get() map { response =>
+      (response.status, response) match {
+        case (200, response) =>
+          extractHashCustomFieldIdFrom(response.json)
+        case (status, response) =>
+          throw new Exception("Problem retrieving the 'Hash' custom field ('%s'): %s" format (status, response.body))
+      }
+    }
+
+  private def extractHashCustomFieldIdFrom(json: JsValue) = {
+    val hashCustomField =
+      json.as[Seq[JsValue]]
+        .filter { field =>
+          (field \ "name").as[String] == configuration.hashCustomFieldName
         }
-    Await.result(projectPromise, 10 seconds)
-  }
+        .headOption
+        .getOrElse(throw new Exception("Could not find a field with the name 'Hash'"))
 
-  /**
-   * Determines the correct component id
-   */
-  lazy val componentId = {
-    val components = (project \ "components").as[Seq[JsValue]]
+    (hashCustomField \ "id").as[String]
+  }
 
-    val foundComponents = for {
-      component <- components
-      if ((component \ "name").as[String] == componentName)
-    } yield (component \ "id").as[String]
+  private lazy val project =
+    request("project/%s" format configuration.projectKey)
+      .get() map { response =>
+        (response.status, response) match {
+          case (200, response) => response.json
+          case (status, response) =>
+            throw new Exception("Problem retrieving the 'Hash' custom field ('%s'): %s" format (status, response.body))
+        }
+      }
 
-    foundComponents
-      .headOption
-      .getOrElse(throw new Exception("Problem retrieving the id of the component '%s', component not found" format componentName))
-  }
+  private lazy val componentId =
+    for {
+      project     <- project
+      components  =  (project \ "components").as[Seq[JsValue]]
+      component   =  findComponentIn(components)
+      componentId =  (component \ "id").as[String]
+    } yield componentId
+
+  private def findComponentIn(components: Seq[JsValue]) =
+    components
+      .find { component =>
+        (component \ "name").as[String] == configuration.componentName
+      }
+      .getOrElse {
+        sys.error("Problem retrieving the id of the component '%s', component not found" format configuration.componentName)
+      }
 
-  lazy val projectId = (project \ "id").as[String]
+  private lazy val projectId = project.map(project => (project \ "id").as[String])
 
 
   /**
    * Utility method that is used to perform requests
    */
-  protected def request(relativePath: String) = {
-    val completeUrl = endpoint + relativePath
+  private def request(relativePath: String) = {
+    val completeUrl = configuration.endpoint + relativePath
 
     Logger info ("Complete URL: %s" format completeUrl)
```

Comment:
Zou je zo iets niet in de debug willen zetten? Voor mij gevoel zou je dit alleen gebruiken als je iets wilt debuggen?